tests/validation/test_highlights.py

```diff
@@ -0,0 +1,366 @@
+
+    def test_no_property_value_highlight(self):
+        response = TestObject()
+        response.is_success = True
+        _json = copy.deepcopy(pytest.JSON)
+        changing_metric_value_per_collection = LongCollectionBundle(5, 25)
+
+        for collection in range(10):
+            response.text = json.dumps(_json)
+            collection_bundle = CollectionBundle(pytest.REQUEST, copy.deepcopy(response), pytest.DURATION,
+                                                 pytest.CONTAINER_STATISTICS)
+            collection_bundle.collection_number = collection
+            changing_metric_value_per_collection.add(collection_bundle)
+
+        highlight = highlight_property_value_growth(changing_metric_value_per_collection.long_collection_statistics)
+        assert len(highlight.messages) == 0
+
+    def test_relationship_highlight(self):
+        response = TestObject()
+        response.is_success = True
+        _json = copy.deepcopy(pytest.JSON)
+        increasing_number_of_property_values_per_collection = LongCollectionBundle(5, 25)
+        parent_object = {
+            "events": [],
+            "key": {
+                "adapterKind": "HighlightsMP",
+                "identifiers": [],
+                "name": "Parent",
+                "objectKind": "Test Parent"
+            },
+            "metrics": [],
+            "properties": []
+        }
+        parent_relationship = {
+            "adapterKind": "HighlightsMP",
+            "identifiers": [],
+            "name": "Parent",
+            "objectKind": "Test Parent"
+        }
+        for collection in range(10):
+            response.text = json.dumps(_json)
+            collection_bundle = CollectionBundle(pytest.REQUEST, copy.deepcopy(response), pytest.DURATION,
+                                                 pytest.CONTAINER_STATISTICS)
+            collection_bundle.collection_number = collection
+            increasing_number_of_property_values_per_collection.add(collection_bundle)
+
+            child_object = copy.deepcopy(parent_object)
+            child_object["key"]["name"] = f"child {collection % 2}"
+            child_object["key"]["objectKind"] = "Test Child"
+            child_relationship = [{
+                "adapterKind": "HighlightsMP",
+                "identifiers": [],
+                "name": child_object["key"]["name"],
+                "objectKind": "Test Child"
+            }]
+
+            _json["result"] = [parent_object, child_object]
+            _json["relationships"] = [{"parent": parent_relationship, "children": child_relationship}]
+
+        highlight = highlight_relationship_growth(
+            increasing_number_of_property_values_per_collection.long_collection_statistics)
+        assert (ResultLevel.WARNING,
+                "Object of type HighlightsMP::Test Parent displayed relationship growth of 2.81%") in highlight.messages
```

Comment:
We may want to confirm with the platform team how relationships are processed from the API. In the original SDK, it was a problem if relationships changed too much, even if they weren't growing. For example, if a resource is constantly switching between several parents, the number of unique relationships won't change much but if there were enough relationship changes that could be a problem.

Reply:
My read of their response is that this is something we will want to detect.